src/libraries/System.Private.CoreLib/src/System/Number.Dragon4.cs

```diff
@@ -537,5 +561,171 @@ private static unsafe uint Dragon4(ulong mantissa, int exponent, uint mantissaHi
             Debug.Assert(outputLen <= buffer.Length);
             return outputLen;
         }
+
+        private static unsafe Dragon4State Dragon4GetState(ulong mantissa, int exponent, uint mantissaHighBitIdx, bool hasUnequalMargins, int cutoffNumber)
```

Comment:
Note, none of this function is new, git is just confused by my refactor. This is just the first half of the existing Dragon4 code hoisted to a helper function.